llitgi/ViewControllers/ManageTags/ManageTagsViewController.swift

```diff
@@ -0,0 +1,256 @@
+//
+//  ManageTagsViewController.swift
+//  llitgi
+//
+//  Created by Xavi Moll on 22/12/2018.
+//  Copyright © 2018 xmollv. All rights reserved.
+//
+
+import UIKit
+
+private enum Section: Int, CaseIterable {
+    case currentTags = 0
+    case availableTags = 1
+    
+    init(section: Int) {
+        switch section {
+        case 0: self = .currentTags
+        case 1: self = .availableTags
+        default: fatalError("You've messed up.")
+        }
+    }
+    
+    var title: String {
+        switch self {
+        case .currentTags: return L10n.Tags.current
+        case .availableTags: return L10n.Tags.available
+        }
+    }
+}
+
+class ManageTagsViewController: UIViewController {
+    
+    //MARK:- IBOutlets
+    @IBOutlet private var tableView: UITableView!
+    @IBOutlet private var toolBar: UIToolbar!
+    @IBOutlet private var newTagBarButtonItem: UIBarButtonItem!
+    private lazy var cancelBarButtonItem: UIBarButtonItem = {
+        let barButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(cancelTapped(_:)))
+        return barButtonItem
+    }()
+    private lazy var saveBarButtonItem: UIBarButtonItem = {
+        let barButtonItem = UIBarButtonItem(barButtonSystemItem: .save, target: self, action: #selector(saveTapped(_:)))
+        return barButtonItem
+    }()
+    private lazy var loadingButton: UIBarButtonItem = {
+        let loading = UIActivityIndicatorView(style: .gray)
+        loading.startAnimating()
+        return UIBarButtonItem(customView: loading)
+    }()
+    
+    //MARK:- Class properties
+    let item: Item
+    let dataProvider: DataProvider
+    let themeManager: ThemeManager
+    let completed: () -> Void
+    private(set) var currentTags: [Tag] = []
+    private(set) var availableTags: [Tag] = []
+    
+    
+    //MARK:- Lifecycle
+    init(item: Item, dataProvider: DataProvider, themeManager: ThemeManager, completed: @escaping () -> Void) {
+        self.item = item
+        self.dataProvider = dataProvider
+        self.themeManager = themeManager
+        self.completed = completed
+        self.currentTags = item.tags
+        self.availableTags = dataProvider.tags.filter { tag in
+            return !item.tags.contains(where: { $0.name == tag.name} )
```

Comment:
This feels hacky, but making tags conform to `Hashable` was a mess when I tried it. I need to revisit this.